### PR TITLE
Run teleport on ECS

### PR DESCRIPTION
**Pull request by iuriaranda**
This module will deploy Teleport auth and proxy servers on an ECS cluster.
It will also create an ELB in front of proxy.

See the module documentation in the Readme for a full description of what it does.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,3 +78,66 @@ module "teleport_bootstrap_script" {
...
+
+### Available variables:
+* [`cluster_name`]: String(required): Name of the cluster.
+* [`domain_name`]: String(required): Domain name of where we want to reach our cluster. Example can be `company.com`
+* [`nlb_private_arn`]: String(required): ARN for the private NLB to create a listener for the Node `auth` containers
+* [`vpc_id`]: String(required): VPC ID of where we want to deploy Teleport in
+* [`aws_region`]: String(optional): AWS region where the CloudWatch logs are located. Defaults to eu-west-1
+* [`cpu`]: Integer(optional): The number of CPU units used by the task. It can be expressed as an integer using CPU units, for example 1024, or as a string using vCPUs, for example 1 vCPU or 1 vcpu, in a task definition but will be converted to an integer indicating the CPU units when the task definition is registered. Defaults to 128.
+* [`dynamodb_table`]: String(optional): Which DynamoDB table does teleport need, teleport will create this table for you. You don't need to define anything in Terraform. Defaults to main.teleport
+* [`dynamodb_region`]: String(optional): In which region does the DynamoDB table need to be created. Defaults to eu-west-1
```
Comment (SamClinckspoor) [inline README.md]: will we ever need this to be different to `aws_region`?

File: README.md [hunk 2]
```diff
@@ -78,3 +78,66 @@ module "teleport_bootstrap_script" {
...
+* [`environment`]: String(optional): Environment where this node belongs to, will be the third part of the node name. Defaults to ''
```
Comment (SamClinckspoor) [inline README.md]: can't you fetch this from terraform.workspace? afaik you have access to that inside the module.

File: README.md (hunk 1 above)
Comment (MattiasGees) [reply README.md]: Maybe we can do it like here? https://github.com/skyscrapers/terraform-kubernetes/blob/master/base/main.tf#L118

File: README.md (hunk 2 above)
Comment (iuriaranda) [reply README.md]: I wouldn't do it like that, as we would be limiting this module to a specific use case. We do map environments to workspaces, but not everybody does, even we don't do it everywhere.
We should keep our terraform modules as generic as possible imo

**Review by SamClinckspoor — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (iuriaranda) [reply README.md]: fixed

**Review by simonrondelez — APPROVED** [review]

**Review by duboisph — APPROVED** [review]

**Pull request merged**